src/token.sw

```diff
@@ -25,3 +27,72 @@ pub fn force_transfer(amount: u64, asset_id: ContractId, contract_id: ContractId
         tr r3 r1 r2;
     }
 }
+
+/// Transfer `amount` coins of type `asset_id` to address `recipient`.
+pub fn transfer_to_output(amount: u64, asset_id: ContractId, recipient: Address) {
+    // note: if tx format changes, the magic number "56" must be changed !
+    // TransactionScript outputsCount has a 56 byte(7 words * 8 bytes) offset
+    // Transaction Script: https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#transactionscript
+    // Output types: https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/tx_format.md#output
+    const OUTPUT_LENGTH_LOCATION = 56;
+    const OUTPUT_VARIABLE_TYPE = 4;
+
+    // get length of outputs from TransactionScript outputsCount:
+    let length: u8 = asm(outputs_length, outputs_length_ptr: OUTPUT_LENGTH_LOCATION) {
+        lb outputs_length outputs_length_ptr i0;
+        outputs_length: u8
+    };
+    // maintain a manual index as we only have `while` loops in sway atm:
+    let mut index: u8 = 0u8;
+    let mut outputIndex = 0;
+    let mut output_found = false;
+
+    // If an output of type `OutputVariable` is found, check if its `amount` is zero.
+    // As one cannot transfer zero coins to an output without a panic, a variable output with a value of zero is by definition unused.
+    while index < length {
+        let output_start = asm(n: index, offset) {
```

Comment:
Note: I separated the use of the `xos` opcode into its own asm block here to avoid calling it twice per iteration.
this way, `output_start` can be reused below, passing it into both the `type` and `amount` asm blocks